plugins/droid/droid-codec.cpp

```diff
@@ -568,7 +602,19 @@ bool DroidVideoDecoder::createCodec()
         droid_media_codec_set_callbacks(m_codec, &cb, this);
     }
 
-    {
+    if (!m_disable_media_buffers) {
+      m_buffer_queue = droid_media_codec_get_buffer_queue (m_codec);
```

Comment:
I didn't follow the logic through entirely, but is there a danger here that `m_buffer_queue` could be dirty from previous calls to `DroidVideoDecoder::init`?
```
DroidVideoDecoder decoder();
setenv("GECKO_CAMERA_DROID_NO_MEDIA_BUFFER", "0");
decoder.init(metadata); // sets m_disable_media_buffers to false
DroidVideoDecoder::createCodec(); // sets m_buffer_queue to non-null
...
setenv("GECKO_CAMERA_DROID_NO_MEDIA_BUFFER", "1");
decoder.init(metadata); // sets m_disable_media_buffers to true
DroidVideoDecoder::createCodec(); // logs "Using media buffers" in spite of the setting
```
Maybe this isn't possible in practice.

Reply:
`m_buffer_queue` is cleaned in `DroidVideoDecoder::stop()`, so it is not possible in normal flow. I changed the code anyway, to always initialize the variable.

Reply:
Thanks for clarifying, I see what you mean. Thanks for updating the code anyway.